src/Swarrot/Processor/XDeath/XDeathMaxLifetimeProcessor.php

```diff
@@ -0,0 +1,186 @@
+     */
+    public function __construct(
+        ProcessorInterface $processor,
+        MessageProviderInterface $messageProvider,
+        \Closure $callback,
+        LoggerInterface $logger = null
+    )
+    {
+        $this->processor = $processor;
+        $this->messageProvider = $messageProvider;
+        $this->callback = $callback;
+        $this->logger = $logger ?: new NullLogger();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setDefaultOptions(OptionsResolver $resolver)
+    {
+        $resolver
+            ->setDefaults(array(
+                'x_death_max_lifetime' => 3600,
+                'x_death_max_lifetime_log_levels_map' => array(),
+                'x_death_max_lifetime_fail_log_levels_map' => array(),
+            ))
+            ->setAllowedTypes('x_death_max_lifetime', 'int')
+            ->setAllowedTypes('x_death_max_lifetime_log_levels_map', 'array')
+            ->setAllowedTypes('x_death_max_lifetime_fail_log_levels_map', 'array');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function process(Message $message, array $options)
+    {
+        try {
+            return $this->processor->process($message, $options);
+        } catch (\Throwable $e) {
+            return $this->handleException($e, $message, $options);
+        } catch (\Exception $e) {
+            return $this->handleException($e, $message, $options);
+        }
+    }
+
+    /**
+     * @param $exception
+     * @param Message $message
+     * @param array $options
+     *
+     * @return mixed
+     */
+    private function handleException($exception, Message $message, array $options)
+    {
+        $headers = $message->getProperties();
+        if (isset($headers['headers']['x-death'], $options['x_death_max_lifetime'])) {
+            $xDeathHeaders = $headers['headers']['x-death'];
+            // PhpAmqpLib compatibility
+            if (class_exists('PhpAmqpLib\Wire\AMQPArray') && $headers['headers']['x-death'] instanceof AMQPArray) {
+                $xDeathHeaders = $headers['headers']['x-death']->getNativeData();
+            }
+            $queueName = $this->messageProvider->getQueueName();
+            $queueXDeathHeader = null;
+            foreach ($xDeathHeaders as $xDeathHeader) {
+                if (isset($xDeathHeader['queue']) && $xDeathHeader['queue'] === $queueName) {
+                    $queueXDeathHeader = $xDeathHeader;
+                    break;
+                }
+            }
+
+            if (null === $queueXDeathHeader) {
+                $this->logException(
+                    $exception,
+                    sprintf(
+                        '[XDeathMaxLifetime] No x-death header found for queue name "%s". Do nothing.',
+                        $queueName
+                    ),
+                    $options['x_death_max_lifetime_fail_log_levels_map']
+                );
+            }
+
+            if (isset($queueXDeathHeader, $queueXDeathHeader['time'])) {
```

Comment:
use `null !== $queueXDeathHeader` instead of putting it inside the `isset`. It makes the code more robust by forbidding undefined variable.